qcodes/data/data_set.py

```diff
@@ -695,9 +703,31 @@ def get_arrays(
             action_indices: All array action indices must start with the provided
                 indices.
             set_arrays: Whether to include setpoint arrays
+            data_group: Optional data group that the array should belong to
         """
         arrays = sorted(self.arrays.values(), key=lambda arr:arr.action_indices)
 
+        # Filter by data group
+        if data_group is not None:
+            if 'data_groups' not in self.metadata:
+                warn('Metadata does not contain data groups')
+            else:
+                valid_idxs = [
+                    tuple(key) for key, val in self.metadata['data_groups']
+                    if data_group in val
+                ]
+                if not valid_idxs:
+                    warn(f'Data group {data_group} not found in metadata')
+                else:
+                    filtered_arrays = []
+                    for valid_idx in valid_idxs:
+                        for arr in arrays:
+                            if arr in filtered_arrays:
+                                continue
+                            elif arr.action_indices[:len(valid_idx)] == valid_idx:
+                                filtered_arrays.append(arr)
```

Comment:
This line was super confusing to me until I executed the code myself, because `valid_idx` itself is actually a tuple of indices, rather than a single index. Is there an easy way to clarify this in the code?